components/backup-stream/src/router.rs

```diff
@@ -1144,16 +1267,17 @@ impl StreamTaskHandler {
         merged_file_info.set_max_ts(max_ts);
         merged_file_info.set_min_ts(min_ts);
         merged_file_info.set_min_resolved_ts(min_resolved_ts.unwrap_or_default());
+        let (unpin_reader, file_encryption_info_vec, hasher_vec) = self
+            .build_unpin_reader_with_encryption_if_needed(data_files_open)
```

Comment:
I think maybe it is unnecessary to do `decrypt + encrypt` from `encrypted tempfiles` to `merged files with encrypted tempfiles`  in the internal operation.
In the original code:
1. writes kv into some log files.   `[ plaintext log file 1 ] [ plaintext log file 2 ] ..  [ plaintext log file N]`
2. writes log files into one merge files in sequence. `{ merged file: [ plaintext log file 1 ] [ plaintext log file 2 ] ..  [ plaintext log file N] }`

However, now:
1. writes kv into some log files.   `[ encrypted log file 1 ] [ encrypted log file 2 ] ..  [ encrypted log file N]`
2. decrypt the files and push it into `data_files_open`.  `[ plaintext log file 1 ] [ plaintext log file 2 ] ..  [ plaintext log file N]`
3. encrypt the files and writes them into one merge files in sequence. `{ merged file: [ encrypted log file 1 ] [ encrypted log file 2 ] ..  [ encrypted log file N] }`

Reply:
Maybe we can remove the decryption operation in the `self.temp_file_pool.open_raw_for_read`, so that we doesn't need to encrypt again.

Reply:
yes that's a valid point, I was thinking about the same at first as well. 

It's a bit tricky in the implementation as the local files are encrypted by the TiKVKeyManager, let me see if I can make it work, otherwise I will create an issue and address after the release.

Reply:
after poking around I feel like to make it work we have to have one KeyManager per task so can keep the local tmp file encryption method same as what user specifies. Looking at the code currently KeyManager is a single instance per TiKV, seems a bit heavy to create a new one for every backup task? Please let me know your thoughts on this!